publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -382,6 +386,28 @@ public List<LogEntry> getLogEntriesForResource(Resource resource) {
                     .toList();
     }
 
+    public void persistFile(FileEntry fileEntry) {
+        var dao = fileEntry.toDao().createInsertionTransactionRequest();
+        sendTransactionWriteRequest(dao);
+    }
+
+    public Optional<FileEntry> fetchFile(FileEntry fileEntry) {
+        var primaryKey = fileEntry.toDao().primaryKey();
+        var result = client.getItem(new GetItemRequest().withTableName(tableName).withKey(primaryKey));
+        return Optional.ofNullable(result.getItem())
+                   .map(FileDao::fromDynamoFormat)
+                   .map(FileEntry::fromDao);
+    }
```

Comment:
I think it is nice to return Optional here and do not throw ApiGateway NotFoundException as we do some other places, so we handle specific behavior in adapters/ services.